test/prov_config_test.c

```diff
@@ -85,7 +114,20 @@ int setup_tests(void)
     if (!TEST_ptr(recurseconfigfile = test_get_argument(1)))
         return 0;
 
+    if (!TEST_ptr(pathedconfig = test_get_argument(2)))
+        return 0;
+
     ADD_TEST(test_recursive_config);
     ADD_TEST(test_double_config);
+#if !defined(OPENSSL_SYS_WINDOWS) && !defined(OPENSSL_SYS_MACOSX) && !defined(NO_PROVIDER_MODULE)
```

Comment:
Hm... are there possibly other platforms where this won't work? We have DSO_EXTENSION macro defined in `include/crypto/dso_conf.h`. Perhaps we could use that?

Reply:
We could use DSO_EXTENSION I believe, but that (correct me if I'm wrong) would require picking that up from inside the perl recepie for the test and rewriting the config file, so that the modulepath setting was correct for it.

Reply:
I did not mean to use it to mangle the config file but to TEST_skip() the testcase if it is something else than `".so"`

Reply:
Ah, ok, that makes sense.  I was thinking adjust the config file so we could test on all platforms, but I like that better

Reply:
updated the test to stat the p_test.so module and skip if its not present.  This covers us both in the case where its named with a different extension, or if the build configuration doesn't generate the file at all